src/power_grid_model/core/data_handling.py

```diff
@@ -151,6 +156,44 @@ def create_output_data(
             shape: Union[Tuple[int], Tuple[int, int]] = (batch_size, count)
         else:
             shape = (count,)
-        result_dict[name] = initialize_array(output_type.value, name, shape=shape, empty=True)
+        attributes = output_component_types[name]
+        if attributes is None:
+            result_dict[name] = initialize_array(output_type.value, name, shape=shape, empty=True)
+        elif attributes in [[], set()]:
+            result_dict[name] = {}
+        else:
+            raise NotImplementedError(
+                "Columnar data types are not implemented yet."
+                "output_component_types must be provided with a list or set"
+            )
 
     return result_dict
+
+
+def validate_output_component_types(output_type: OutputType, dict_output_types: _OutputComponentTypeDict):
+    """Checks dict_output_types for any invalid component names and attribute names
+
+    Args:
+        output_type (OutputType): the type of output that the user will see (as per the calculation options)
+        dict_output_types (Dict[ComponentType, Optional[str]]): output_component_types converted to dictionary
```

Comment:
I think the correct typing for `dict_output_types` in the docstring should be `Dict[ComponentType, Optional[Union[Set[str], List[str]]]]`